examples/custom_import_schemes.rs

```diff
@@ -0,0 +1,53 @@
+use deno_core::{anyhow::Error, ModuleSpecifier};
+use rustyscript::{ImportProvider, Runtime};
+
+struct MyImportProvider {
+    example_code: String,
+}
+impl MyImportProvider {
+    fn new(str: &str) -> Self {
+        Self {
+            example_code: String::from(str),
+        }
+    }
+}
+impl ImportProvider for MyImportProvider {
+    fn import(
+            &mut self,
+            specifier: ModuleSpecifier,
+            _referrer: Option<ModuleSpecifier>,
+            _is_dyn_import: bool,
+            _requested_module_type: deno_core::RequestedModuleType,
+        ) -> Result<String, deno_core::anyhow::Error> {
+        if specifier.scheme() == "examplescheme" {
+            // Load code from anywhere 
+            Ok(self.example_code.clone())
+        } else {
+            // Make sure to handle unrecognized schemes as well
+            Err(Error::msg(format!("unrecognized scheme for module import: {specifier}")))
```

Comment:
Needing this everywhere is maybe a case for the trait doing Option<Result>> to make it easier for users to fallback to default?

Reply:
I think I misunderstood what you meant when you suggested this before. Will try this out later.

Reply:
> I think I misunderstood what you meant when you suggested this before. Will try this out later.

Sorry 
Option<Result> as the function return

Just an idea not a demand

Reply:
It's a good idea. I think it could benefit readability.